code/game/turfs/closed/minerals.dm

```diff
@@ -183,21 +188,34 @@
 		gets_drilled(give_exp = FALSE)
 
 /turf/closed/mineral/random
-	var/list/mineralSpawnChanceList = list(/obj/item/stack/ore/uranium = 5, /obj/item/stack/ore/diamond = 1, /obj/item/stack/ore/gold = 10,
-		/obj/item/stack/ore/silver = 12, /obj/item/stack/ore/plasma = 20, /obj/item/stack/ore/iron = 40, /obj/item/stack/ore/titanium = 11,
-		/turf/closed/mineral/gibtonite = 4, /obj/item/stack/ore/bluespace_crystal = 1)
-		//Currently, Adamantine won't spawn as it has no uses. -Durandan
 	var/mineralChance = 13
 
-/turf/closed/mineral/random/Initialize(mapload)
-	if(check_holidays(APRIL_FOOLS))
-		mineralSpawnChanceList[/obj/item/stack/ore/bananium] = 3
+/// Returns a list of the chances for minerals to spawn.
+/// Will only run once, and will then be cached.
+/turf/closed/mineral/random/proc/mineral_chances()
+	return list(
+		/obj/item/stack/ore/bananium = check_holidays(APRIL_FOOLS) ? 3 : 0,
+		/obj/item/stack/ore/bluespace_crystal = 1,
+		/obj/item/stack/ore/diamond = 1,
+		/obj/item/stack/ore/gold = 10,
+		/obj/item/stack/ore/iron = 40,
+		/obj/item/stack/ore/plasma = 20,
+		/obj/item/stack/ore/silver = 12,
+		/obj/item/stack/ore/titanium = 11,
+		/obj/item/stack/ore/uranium = 5,
+		/turf/closed/mineral/gibtonite = 4,
+	)
 
-	mineralSpawnChanceList = typelist("mineralSpawnChanceList", mineralSpawnChanceList)
+/turf/closed/mineral/random/Initialize(mapload)
+	var/static/list/mineral_chances_by_type = list()
 
 	. = ..()
 	if (prob(mineralChance))
-		var/path = pick_weight(mineralSpawnChanceList)
+		var/list/spawn_chance_list = mineral_chances_by_type[type]
+		if (isnull(spawn_chance_list))
+			mineral_chances_by_type[type] = mineral_chances()
+			spawn_chance_list = mineral_chances_by_type[type]
+		var/path = pick_weight(spawn_chance_list)
```

Comment:
if you're going to cache this, may I suggest using expand_weights() and pick() rather then pick_weights()?
it has a memory cost but is significantly faster, and I bet would help here

Reply:
at least well, depending on how hot this actually is

Reply:
Alright